webviz_config/utils/_available_port.py

```diff
@@ -1,18 +1,37 @@
 import os
 import socket
+from typing import Optional
 
 
-def get_available_port() -> int:
+def get_available_port(preferred_port: Optional[int] = None) -> int:
     """Finds an available port for use in webviz on localhost. If a reload process,
     it will reuse the same port as found in the parent process by using an inherited
     environment variable.
+
+    If preferred_port is given, ports in the range [preferred_port, preferred_port + 20)
+    will be tried first, before an OS provided random port is used as fallback.
     """
 
+    def is_available(port):
+        with socket.socket() as sock:
+            try:
+                sock.bind(("localhost", port))
+                return True
+            except OSError:
+                return False
+
     if os.environ.get("WEBVIZ_PORT") is None:
-        sock = socket.socket()
-        sock.bind(("localhost", 0))
-        port = sock.getsockname()[1]
-        sock.close()
+        port = None
+
+        if preferred_port is not None:
+            for port_to_test in range(preferred_port, preferred_port + 20):
+                if port is None and is_available(port_to_test):
+                    port = port_to_test
```

Comment:
Could break loop when successful, but the max 20 iterations and `port is None` check makes the gain limited of course
```suggestion
                    port = port_to_test
                    break
```